### PR TITLE
Rename `checkOnSave` settings to `check`

### PR DESCRIPTION
**Pull request by Veykril**
Now that flychecks can be triggered without saving the setting name doesn't make that much sense anymore. This PR renames it to just `check`, but keeps `checkOnSave` as the enabling setting.

### CODE REVIEW

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
Sounds like a good idea to be consistent about this. I don't think the more obscure name is a problem.

**Comment by Veykril** [comment]
Oh okay, let's not use flycheck publicly if it clashes with terminology from emacs, that does sound like a bad choice. `externalCheck` sounds good, though maybe `check` itself really would be fine as well.

File: crates/rust-analyzer/src/config.rs [hunk 1]
```diff
@@ -1862,35 +1863,30 @@ fn get_field<T: DeserializeOwned>(
...
+        .unwrap_or_else(|| serde_json::from_str(default).unwrap())
 }
 
 fn schema(fields: &[(&'static str, &'static str, &[&str], &str)]) -> serde_json::Value {
-    for ((f1, ..), (f2, ..)) in fields.iter().zip(&fields[1..]) {
-        fn key(f: &str) -> &str {
-            f.splitn(2, '_').next().unwrap()
-        }
-        assert!(key(f1) <= key(f2), "wrong field order: {f1:?} {f2:?}");
-    }
```
Comment (Veykril) [inline crates/rust-analyzer/src/config.rs:1892]: We already assert this via a separate test

**Comment by kylebarron** [comment]
Just a note that I think this broke any configuration users had already set based on `checkOnSave`. I had set `checkOnSave.command` to `clippy` and that didn't automatically get updated to `check.command` when updating to this release

**Comment by lnicola** [comment]
@kylebarron the server still understands the old preference, even if it shows up as unused in Code.

**Comment by kylebarron** [comment]
For me when I updated to the latest rust-analyzer version, clippy-on-save stopped working entirely 🤷‍♂️. It wasn't until I switched `checkOnSave.command` to `check.command` until it started working again. It's also possible I did something wrong though!

**Comment by lnicola** [comment]
Idle musing: I wonder if we should add a "Run Clippy on crate/workspace" command to encourage users to not set Clippy as the default check command.

**Comment by lnicola** [comment]
@kylebarron FWIW, I can't reproduce that:

image

**Comment by kylebarron** [comment]
🤷‍♂️. I probably did something wrong! sorry for the noise!

**Pull request merged**